version: '3.4'

volumes:
  local_postgres_data: {}
  local_postgres_data_backups: {}

services:
  django: &django
    build:
      context: .
      dockerfile: ./compose/local/django/Dockerfile
    image: lae_observer_local_django
    depends_on:
      - postgres
    volumes:
      - .:/app
    env_file:
      - ./.envs/.local/.django
      - ./.envs/.local/.postgres
    ports:
      - "8000:8000"
    command: /start
    restart: on-failure

  postgres:
    image: postgres:10.5
    volumes:
      - ./var/postgres-data:/var/lib/postgresql/data
    env_file:
      - ./.envs/.local/.postgres
    restart: on-failure
    logging:
      driver: none

  redis:
    image: redis:3.2

  celeryworker:
    <<: *django
    image: lae_observer_local_celeryworker
    restart: on-failure
    depends_on:
      - redis
      - postgres

    ports: []
    command: /start-celeryworker

  flower:
    <<: *django
    image: lae_observer_local_flower
    restart: on-failure
    ports:
      - "5555:5555"
    command: /start-flower